sdk/go/x/auto/git.go

```diff
@@ -21,11 +21,57 @@ import (
 	"github.com/pkg/errors"
 	git "gopkg.in/src-d/go-git.v4"
 	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/http"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/ssh"
 )
 
 func setupGitRepo(ctx context.Context, workDir string, repoArgs *GitRepo) (string, error) {
+	cloneOptions := &git.CloneOptions{
+		URL: repoArgs.URL,
+	}
+
+	if repoArgs.Auth != nil {
+
+		authDetails := repoArgs.Auth
+		// Each of the authentication options are mutually exclusive so let's check that only 1 is specified
+		if (authDetails.SSHPrivateKeyPath != "" && authDetails.PersonalAccessToken != "") ||
+			(authDetails.SSHPrivateKeyPath != "" && authDetails.Password != "") ||
```

Comment:
This condition currently causes an err for a valid case if providing a passphrase for the SSH key when used below in L46.

I believe we'd like this line to error if the username is also set.

Reply:
You are correct - it should be using username in this condition not the password